.circleci/config.yml

```diff
@@ -57,13 +57,27 @@ default-context: &default-context
   context:
     - org-global
 
+test-react-18-context: &test-react-18-context
+  <<: *default-context
+  react-version: ^18.0.0
+  filters:
+    branches:
+      only:
+        - master
+        - /^v.*\.x$/
+        - next
```

Comment:
How does it know it should trigger the `test-react-18` workflow?

Reply:
When pipeline is triggered on a filtered branch.
I have tested it on this branch by adding `renovate/react-major`. 😉 
I assume this will cause every CI to run on those branches to additionally test on React 18. 🤔 

Would you rather go for a nightly cron job?

My thought process:
- Adding it to regular CI runs is a bit too costly
- Running only nightly might provide a subpar feedback cycle and miss the accountability (cron triggered job will probably not have the commit author and a failure notification could be missed)
- Running on every master push (PR merge) seemed like the best compromise. 🤔

Reply:
> Would you rather go for a nightly cron job?

No, I think running on selected branches is optimal solution.
It's just that I don't quite get how `test-react-18-context` and `test-react-18` are related.
If the filters in the `test-react-18-context` are passed, does it automatically run all jobs that are tied to the `test-react-18-context`?

Reply:
> It's just that I don't quite get how `test-react-18-context` and `test-react-18` are related.

`test-react-18-context` is basically a reusable "variable" to avoid duplicating the same config for every test.
Without this, every job after L436 would get the following diff:
```diff
-<<: *test-react-18-context
+<<: *default-context
+react-version: ^18.0.0
+filters:
+   branches:
+     only:
+       - master
+       - /^v.*\.x$/
+       - next
```